app/code/community/Lesti/Fpc/Helper/Data.php

```diff
@@ -258,4 +259,30 @@ public function getContentType(\Mage_Core_Controller_Response_Http $response)
 
         return 'text/html; charset=UTF-8';
     }
+
+    /**
+     * Transform array parameter to string to avoid error and make correct validation in regexp pattern
+     *
+     * @param mixed $param Input parameter
+     *
+     * @return mixed
+     */
+    protected function _paramToString($param)
+    {
+        if (is_array($param)) {
+            $param = array_reduce(
+                $param,
+                function ($carry, $item) {
```

Comment:
Now I understand what you mean.
In this case the only one solution - walk recursive throw array and apply regexp to every entry.
And if no matches than no cache.
Maybe this not covers all situations, but this is a best that we can do, I think